src/shared/components/NotFound.tsx

```diff
@@ -134,63 +124,41 @@ const NotFoundNew: FC = () => (
   </AppWrapper>
 )
 
-const NotFoundOld: FC = () => (
-  <div className="container-fluid" data-testid="not-found">
-    <div className="panel">
-      <div className="panel-heading text-center">
-        <h1 className="deluxe">404</h1>
-        <h4>Bummer! We couldn't find the page you were looking for</h4>
-      </div>
-    </div>
-  </div>
-)
+const NotFound: FC = () => {
+  const [isFetchingOrg, setIsFetchingOrg] = useState(false)
+  const location = useLocation()
+  const history = useHistory()
+  const org = useSelector(getOrg)
 
-class NotFound extends Component<Props> {
-  state = {
-    isFetchingOrg: false,
-  }
-
-  async componentDidMount() {
-    if (isFlagEnabled('deepLinking')) {
-      let org = this.props?.org
-
-      if (!org) {
-        this.setState({isFetchingOrg: true})
-        org = await fetchOrg()
-      }
-
-      const deepLinkingMap = buildDeepLinkingMap(org)
-
-      if (deepLinkingMap.hasOwnProperty(this.props.location.pathname)) {
-        event('deeplink', {from: this.props.location.pathname})
-        this.props.history.replace(deepLinkingMap[this.props.location.pathname])
-        return
-      }
-      this.setState({isFetchingOrg: false})
+  const handleDeepLink = useCallback(async () => {
+    if (!org) {
+      setIsFetchingOrg(true)
+      await fetchOrg()
     }
-  }
 
-  render() {
-    if (this.state.isFetchingOrg) {
-      // don't render anything if this component is actively fetching org id
-      // this prevents popping in a 404 page then redirecting
-      return null
+    const deepLinkingMap = buildDeepLinkingMap(org)
+
+    if (deepLinkingMap.hasOwnProperty(location.pathname)) {
+      event('deeplink')
```

Comment:
Do we want to include the pathname in event?

```suggestion
      event('deeplink', {from: location.pathname})
```